aequilibrae/project/field_editor.py

```diff
@@ -0,0 +1,155 @@
+        self._table_fields = [x[1] for x in dt if x[1] != 'ogc_fid']
+
+    def remove(self, field_name: str) -> None:
+        pass
+
+    def save(self) -> None:
+        """Saves any field descriptions which my have been changed to the database"""
+
+        qry = 'update attributes_documentation set description="{}" where attribute="{}" and name_table="{}"'
+        for key, val in self._original_values.items():
+            new_val = self.__dict__[key]
+            if new_val != val:
+                self.__run_query_commit(qry.format(new_val, key, self._table))
+                logger.info(f'Metadata for field {key} on table {self._table} was updated to {new_val}')
+
+    def all_fields(self) -> List[str]:
+        """Returns the list of fields available in the database"""
+        return list(self._original_values.keys())
+
+    def _check_completeness(self) -> None:
+        raw_fields = self._table_fields
+
+        if self._table == 'links':
+            fields = []
+            for field in raw_fields:
+                if field[-3:] == '_ab':
+                    if field[:-3] + '_ba' in raw_fields:
+                        fields.append(field[:-2])
+                    else:
+                        fields.append(field)
+                elif field[-3:] == '_ba':
+                    if field[:-3] + '_ab' in raw_fields:
+                        continue
+                    else:
+                        fields.append(field)
+                else:
+                    fields.append(field)
+        else:
+            fields = raw_fields
+
+        for field in fields:
+            if field not in self._original_values.keys():
+                self.__adds_to_attribute_table(field, 'not provided')
+
+        original_fields = list(self._original_values.keys())
+        for field in original_fields:
+            if field not in fields:
+                qry = f'DELETE FROM attributes_documentation where attribute="{field}" and name_table="{self._table}"'
+                self.__run_query_commit(qry)
+                del self.__dict__[field]
+                del self._original_values[field]
+
+    def __adds_to_attribute_table(self, attribute_name, attribute_value):
+        self.__dict__[attribute_name] = attribute_value
+        self._original_values[attribute_name] = attribute_value
+        qry = 'insert into attributes_documentation VALUES(?,?,?)'
+        vals = (self._table, attribute_name, attribute_value)
+        self.__run_query_commit(qry, vals)
+
+    def __run_query_fetch_all(self, qry: str):
+        conn = database_connection()
+        curr = conn.cursor()
+        curr.execute(qry)
+        dt = curr.fetchall()
+        conn.close()
+        return dt
+
+    def __run_query_commit(self, qry: str, values=None) -> None:
+        conn = database_connection()
+        if values is None:
+            conn.execute(qry)
+        else:
+            conn.execute(qry, values)
+        conn.commit()
+        conn.close()
```

Comment:
Feel like this should be in some utility class

Reply:
What do you mean?